exchanges/deribit/deribit_test.go

```diff
@@ -0,0 +1,3725 @@
+package deribit
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/common/key"
+	"github.com/thrasher-corp/gocryptotrader/config"
+	"github.com/thrasher-corp/gocryptotrader/core"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/fundingrate"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/futures"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/sharedtestvalues"
+	"github.com/thrasher-corp/gocryptotrader/portfolio/withdraw"
+)
+
+// Please supply your own keys here to do authenticated endpoint testing
+const (
+	apiKey    = ""
+	apiSecret = ""
+
+	canManipulateRealOrders   = false
+	canManipulateAPIEndpoints = false
+	btcPerpInstrument         = "BTC-PERPETUAL"
+	useTestNet                = false
+)
+
+var (
+	d                                                                                                            = &Deribit{}
+	futuresTradablePair, optionsTradablePair, optionComboTradablePair, futureComboTradablePair, spotTradablePair currency.Pair
+	fetchTradablePairChan                                                                                        chan struct{}
+	tradablePairsFetchedStatusLock                                                                               = sync.Mutex{}
+	tradablePairsFetched                                                                                         bool
+)
+
+func TestMain(m *testing.M) {
+	d.SetDefaults()
```

Comment:
This can take advtange of the internal test exchange Setup helper, like:

```
d = new(Deribit)
if err := testexch.Setup(d); err != nil {
 	log.Fatal(err)
 }
 
 ```
 
 To rid a lot of the TestMain boilerplate